### PR TITLE
Add objectives to tutorial page

### PR DESCRIPTION
**Pull request by benjimarshall**
## Changes
- Install the Html Agility Pack NuGet package.
- Add tutorial objective parsing, to the view model, and display it in
the Tutorial view.
- Add tests for this view model parsing method.

## Testing
Add view model unit tests, test in Chrome, Firefox, Edge and IE11, using a screen reader and large zoom.

### Collapsed objectives
no_objectives

### Expanded objectives
objectives_expanded

### A tutorial without any objectives
objectives_collapsed

### CODE REVIEW

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/TutorialViewModelTests.cs [hunk 1]
```diff
@@ -160,5 +160,66 @@ public void Tutorial_should_not_recommend_a_tutorial_when_score_is_max()
             // Then
             tutorialViewModel.TutorialRecommendation.Should().Be("optional");
         }
+
+        [Test]
+        public void Tutorial_parsed_objectives_are_null_when_objectives_are_null()
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/TutorialViewModelTests.cs:165]: This test name doesn't quite read like a sentance. Maybe something like `Tutorial_handles_null_objectives`? 🙂

File: DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs [hunk 2]
```diff
@@ -35,5 +40,21 @@ private static string ParseMinutes(int minutes)
...
+                return null;
+            }
+
+            var doc = new HtmlDocument();
+            doc.LoadHtml(objectives);
+
+            var body = doc.DocumentNode.SelectNodes("//body");
+
+            return body?.FirstOrDefault()?.InnerHtml
+                   ?? WebUtility.HtmlDecode(objectives);
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs:57]: Sorry, why do we need to call `HtmlDecode` here? 🙂

File: DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs [hunk 3]
```diff
@@ -35,5 +40,21 @@ private static string ParseMinutes(int minutes)
...
+            {
+                return null;
+            }
+
+            var doc = new HtmlDocument();
+            doc.LoadHtml(objectives);
+
+            var body = doc.DocumentNode.SelectNodes("//body");
+
+            return body?.FirstOrDefault()?.InnerHtml
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs:56]: Why the `FirstOrDefault`? Don't we just want everything inside the body? I was imagining something like this https://html-agility-pack.net/from-string but with inner html instead of outer (I assume that will remove the body tags), although I haven't tried that 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs (hunk 3 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs]: So this produces a list of bodies, and I'm just taking the first body and returning its contents. Select Single node implements the `FirstOrDefault` for us so I've switched over to using that in 3ff95e1 🙂

File: DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs (hunk 2 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web/ViewModels/LearningMenu/TutorialViewModel.cs]: I'm not sure if I need this actually. It converts HTTP strings into normal character strings (eg. from `&gt;` to `>`) but now that I think of it I doubt it will be stored like that in the database (and I can't find any examples). Removed in 3ff95e1 🙂

File: DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/TutorialViewModelTests.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/TutorialViewModelTests.cs]: That's much better, thank you! Changed in 3ff95e1 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂
